blivet/devicelibs/stratis.py

```diff
@@ -0,0 +1,96 @@
+#
+# stratis.py
+# stratis functions
+#
+# Copyright (C) 2020  Red Hat, Inc.  All rights reserved.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Author(s): Vojtech Trefny <vtrefny@redhat.com>
+#
+
+import gi
+gi.require_version("GLib", "2.0")
+
+from gi.repository import GLib
+
+from ..errors import StratisError
+from ..size import Size
+from ..static_data import stratis_info
+from .. import safe_dbus
+
+
+STRATIS_SERVICE = "org.storage.stratis2"
+STRATIS_PATH = "/org/storage/stratis2"
+STRATIS_POOL_INTF = STRATIS_SERVICE + ".pool"
+STRATIS_FILESYSTEM_INTF = STRATIS_SERVICE + ".filesystem"
+STRATIS_BLOCKDEV_INTF = STRATIS_SERVICE + ".blockdev"
+STRATIS_PROPS_INTF = STRATIS_SERVICE + ".FetchProperties"
+STRATIS_MANAGER_INTF = STRATIS_SERVICE + ".Manager"
+
+
+STRATIS_FS_SIZE = Size("1 TiB")
```

Comment:
Can we expect this will be always 1 TiB? We need the size just for the initial device scan, the size is later updated from sysfs.

Reply:
Currently, you can expect this to always start as 1 TiB. It will grow if the physical storage available is greater than 1 TiB and the filesystem fills up. Stratis automatically extends the filesystem size when it fills up. Currently, resizing will double the filesystem size.